qiskit_addon_sqd/fermion.py

```diff
@@ -144,8 +148,13 @@ def solve_fermion(
     )
 
     # Calculate the avg occupancy of each orbital
-    dm1 = myci.make_rdm1s(sci_vec, norb, (num_up, num_dn))
-    avg_occupancy = [np.diagonal(dm1[0]), np.diagonal(dm1[1])]
+    dm1s = myci.make_rdm1s(sci_vec, norb, (num_up, num_dn))
+    avg_occupancy = [np.diagonal(dm1s[0]), np.diagonal(dm1s[1])]
+
+    # Calculate energy from RDMs
+    dm1 = myci.make_rdm1(sci_vec, norb, (num_up, num_dn))
+    dm2 = myci.make_rdm2(sci_vec, norb, (num_up, num_dn))
+    e_sci = np.einsum("pr,pr->", dm1, hcore) + 0.5 * np.einsum("prqs,prqs->", dm2, eri)
```

Comment:
Getting the 1- and 2-RDMs is independently useful. I wonder if we should add a method to `SCIState` to compute them.

Reply:
I'm not opposed to putting a helper somewhere to get these. Can discuss in a future issue.